tests/td_test.c

```diff
@@ -113,6 +113,114 @@ MU_TEST(test_quantile_interpolations) {
     td_free(t);
 }
 
+MU_TEST(test_trimmed_mean_simple) {
+    /* Used numpy to check results validity
+     import numpy as np
+     from scipy import stats
+     x = [5,5,5,10,15,15,15]
+     np.mean(x)
+     10.0
+     stats.trim_mean(x, 0.0)
+     10.0
+     */
+    td_histogram_t *t = td_new(100);
+    mu_assert(t != NULL, "created_histogram");
+    mu_assert_double_eq(0, t->unmerged_weight);
+    mu_assert_double_eq(0, t->merged_weight);
+    //    stats.trim_mean([], 0.49)
+    //    nan
+    mu_assert_double_eq(NAN, td_trimmed_mean_symmetric(t, .49));
+    mu_assert_double_eq(NAN, td_trimmed_mean(t, 0.49, 0.51));
+    td_add(t, 5.0, 1);
+    // with one data point, all quantiles lead to Rome
+    // stats.trim_mean(x, 0.49)
+    mu_assert_double_eq(5, td_trimmed_mean_symmetric(t, .49));
+    mu_assert_double_eq(5, td_trimmed_mean(t, 0.49, 0.51));
+    // stats.trim_mean(x, 0.1)
+    // 5.0
+    mu_assert_double_eq(5, td_trimmed_mean_symmetric(t, .1));
+    mu_assert_double_eq(5, td_trimmed_mean(t, 0.1, 0.9));
+    // 5.0
+    // stats.trim_mean(x, 0.0)
+    mu_assert_double_eq(5, td_trimmed_mean_symmetric(t, .0));
+    mu_assert_double_eq(5, td_trimmed_mean(t, 0.0, 1.0));
+    // 5.0
+    td_add(t, 5.0, 2);
+    mu_assert_double_eq(5, td_trimmed_mean_symmetric(t, .0));
+    mu_assert_double_eq(5, td_trimmed_mean(t, 0.0, 1.0));
+    td_add(t, 10.0, 1);
+    td_add(t, 15.0, 3);
+    //    stats.trim_mean(x, 0.0)
+    //    10.0
+    mu_assert_double_eq(10, td_trimmed_mean_symmetric(t, .0));
+    mu_assert_double_eq(10, td_trimmed_mean(t, 0.0, 1.0));
+    // trimmed mean and mean should lead to 10 in here
+    //    stats.trim_mean(x, 0.1)
+    //    10.0
+    mu_assert_double_eq(10, td_trimmed_mean_symmetric(t, .1));
+    mu_assert_double_eq(10, td_trimmed_mean(t, .1, .9));
+    // trimmed mean and mean should lead to 10 in here
+    //    stats.trim_mean(x, 0.25)
+    //    10.0
+    mu_assert_double_eq(10, td_trimmed_mean_symmetric(t, .25));
+    mu_assert_double_eq(10, td_trimmed_mean(t, .25, .75));
```

Comment:
add a test for asymmetric measurements. Ex. (0.1, 0.75).

Reply:
Fixed in the latest commit